Description of Dataset and the Generation Process (#5)
greenelab/text_mined_hetnet_manuscript: content/03.methods.md

# Materials and Methods

## Dataset

> The dataset used for this project is called Pubtator [@doi:10.1093/nar/gkt441], which is an open source repository that contains pre-tagged medline abstracts.

Reviewer: ```suggestion
We used PubTator [@doi:10.1093/nar/gkt441] as input to our analysis.
PubTator provides medline abstracts that have been annotated with well-established entity recognition tools including (X, Y, Z, ZZ, ZZZ...).
```

> It was downloaded on June 30, 2017 and contained 10,775,748 different abstracts.

Reviewer: ```suggestion
We downloaded PubTator on June 30, 2017, at which point it contained 10,775,748 abstracts.
```

> After obtaining these abstracts, we filtered out mention tags that were not contained in hetionet.

Reviewer: ```suggestion
We filtered out mention tags that were not contained in hetionet.
```

Reviewer: As a note -what is hetionet? I think you need a section in your M&M on it.

> Following the tag filtering step, we used the Stanford CoreNLP parser [@doi:10.3115/v1/P14-5010] to do part of speech (POS) tagging and dependency tree generation.

Reviewer: ```suggestion
We then used the Stanford CoreNLP parser [@doi:10.3115/v1/P14-5010] to tag parts of speech and generate dependency trees.
```

> Treating sentences with only a single mention as negative, we extracted sentences with two or more mentions, termed candidate sentences. 

Reviewer: ```suggestion
We extracted sentences with two or more mentions, termed candidate sentences, and considered sentences with only a single mention to be negatives.
```

Reviewer: I don't understand if this means you removed sentences that had only a single mention, or if they were included but used as negatives for ML somehow.

> After obtaining candidate sentences, we did a stratified sort of each candiadte into a training set, tuning set and a testing set (shown in Table 1).

Reviewer: ```suggestion
We performed a stratified sort (STRATIFIED BY?) to produce a training set, tuning set and a testing set (shown in Table 1).
```

> Each sentence was sorted based on the present mention pair.

Reviewer: I see that this goes with above. Best to work it in there, but as written I also don't quite get it. You stratified sorted by entity or by pairs of entities?

> To obtain a ground truth set for each given relationship, we hand labeled five hundred to a thousand candidate sentences (shown in parenthesis in Table 1).

Reviewer: ```suggestion
We hand labeled five hundred to a thousand candidate sentences of each relationship to obtain to obtain a ground truth set (Table 1).
```

> Table 1. Statistics of Candidate Sentences

Reviewer: ```suggestion
Table 1. Statistics of Candidate Sentences. Numbers in parentheses show the number of positives and negatives that resulted from the hand-labeling process.
```

# Materials and Methods
## Hetionet
Hetionet [@doi:10.7554/eLife.26726] is a large heterogenous network that contains pharmacological and biological information.

> This network depicts information in the form of nodes and edges of different types: nodes that represent biological and pharmacological entities and edges which represent relationships each entity may share with itself or another. 

Reviewer 2: "relationships between entities" may be more clear than "relationships each entity may share with itself or another". The "share with itself" phrasing makes me think of self-loops. There are no self loops in Hetionet v1.0. Try the following Cypher query at https://neo4j.het.io/browser/:

```cypher
MATCH path=(source)-[rel]->(target)
WHERE source = target
RETURN type(rel), count(rel)
```

> Currently, Hetionet contains 47,031 nodes with 11 different data types and 2,250,197 edges that represent 24 different relationship types (Figure 1).

Reviewer 2: You could also just say "Hetionet v1.0" rather than currently.

> Edges in Hetionet were obtained from open databases, such as the GWAS Catalog [@doi:10.1093/nar/gkw1133], DrugBank [@doi:10.1093/nar/gkw1133].

Reviewer 2: missing "and" before DrugBank?

> All of the data in this network comes from publicly available databases, such as GWAS Catalog [@doi:10.1093/nar/gkw1133], DrugBank [@doi:10.1093/nar/gkw1133] and more.

Reviewer: ```suggestion
Edges in Hetionet were obtained from open databases, such as the GWAS Catalog [@doi:10.1093/nar/gkw1133], DrugBank [@doi:10.1093/nar/gkw1133].
```

> For this project, we focus on the following relationship types: disease associates with a gene (DaG), compounds binding to a gene (CbG), gene interacts with gene (GiG) and compound treating a disease (CtD).

Reviewer: ```suggestion
For this project, we analyzed performance over a subset of the Hetionet relationship types: disease associates with a gene (DaG), compounds binding to a gene (CbG), gene interacts with gene (GiG) and compound treating a disease (CtD).
```

> For this project, we analyzed performance over a subset of the Hetionet relationship types: disease associates with a gene (DaG), compounds binding to a gene (CbG), gene interacts with gene (GiG) and compound treating a disease (CtD).

Reviewer 2: "binding" is different tense than others. I'd use "binds" instead.

> ![Hetionet_metagraph](images/figures/metagraph.png)

Reviewer: I think you should use a figure that shows only the relationship types that you actually used. _OR_ a figure like this one, where all of the types you didn't use are "faded" with, say, 70% transparency, into the background.

> Figure 1. A metagraph (schema) of Hetionet where pharmacological, biological and disease entities are represented as nodes and the relationships between them are represented as edges.

Reviewer 2: Great figure. I like the transparency / fading to highlight which Hetionet relationships this study focuses on.

Should the caption discuss the fading (or whatever you want to call it).

See [proper syntax](https://github.com/manubot/rootstock/blob/master/USAGE.md#figures) for Figure caption (goes where alt text usually goes).

> Figure 1. A metagraph (schema) of Hetionet where pharmacological, biological and disease entities are represented as nodes and the relationships between them are represented as edges [@doi:10.7554/eLife.26726].

Reviewer: ```suggestion
Figure 1. A metagraph (schema) of Hetionet where pharmacological, biological and disease entities are represented as nodes and the relationships between them are represented as edges.
```

Reviewer: Removed the ref now b/c the figure has been modified.

## Dataset
We used PubTator [@doi:10.1093/nar/gkt441] as input to our analysis.

> PubTator provides medline abstracts that have been annotated with well-established entity recognition tools including DNorm [@doi:10.1093/bioinformatics/btt474], GeneTUKit [@doi:10.1093/bioinformatics/btr042] Gnorm [@doi:10.1186/1471-2105-12-S8-S5] and a dictionary based look system for compound mentions [@doi:10.1093/database/bas037].

Reviewer 2: Should you mention which entity types each tagger applies to?

medline should be capitalized like MEDLINE

We downloaded PubTator on June 30, 2017, at which point it contained 10,775,748 abstracts. 
Then we filtered out mention tags that were not contained in hetionet.
We used the Stanford CoreNLP parser [@doi:10.3115/v1/P14-5010] to tag parts of speech and generate dependency trees.
We extracted sentences with two or more mentions, termed candidate sentences.

> Each candidates sentence was stratified by co-mention pair to produce a training set, tuning set and a testing set (shown in Table 1).

Reviewer 2: The stratification procedure to preoduct traiin / tune / test is important. Not sure this description is sufficient (unless more details will be provided elsewhere)

Sentences that contain more than one co-mention pair are treated as multiple individual candidates.

> We hand labeled five hundred to a thousand candidate sentences of each relationship to obtain to obtain a ground truth set (Table 1).

Reviewer 2: Probably want to use tablenos automated numbering
https://github.com/manubot/rootstock/blob/master/USAGE.md#tables

Reviewer 2: Also link to the dataset with the hand labels here? Like "(table 1, dataset)"?

| Relationship | Train | Tune | Test |
| :--- | :---: | :---: | :---: |
| Disease Associates Gene | 2.35 M |31K (397+, 603-) | 313K (351+, 649-) |
| Compound Binds Gene | 1.7M | 468K (37+, 463-) | 227k (31+, 469-) |
| Compound Treats Disease | 1.013M | 96K (96+, 404-) | 32K (112+, 388-) |
| Gene Interacts Gene | 12.6M | 1.056M (60+, 440-) | 257K (76+, 424-) |

> Table 1. Statistics of Candidate Sentences. Numbers in parentheses show the number of positives and negatives that resulted from the hand-labeling process.

Reviewer 2: Perhaps a brief comment on split into Train, Tune, Test in caption?
